### PR TITLE
issue: 1004613 TX Post-Send PRM enabled for non-VMAPOLL

### PR DESCRIPTION
**Pull request by olgk**
TX Post Send PRM from VMAPOLL was implemented  with new class _mlx5
derived classes for QP/CP managers. New low level send_to_wire(),
init_sw() were implemented to be virtual and replace mlx5_send(),
mlx5_sq_init().
Rebased to 8.3.5

Signed-off-by: Oleg Kuporosov <olegk@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1863/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1310/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1311/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1865/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1312/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1866/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1313/ for details (Mellanox internal link).

File: src/vma/dev/cq_mgr_mlx5.cpp [hunk 1]
```diff
@@ -419,6 +420,149 @@ int cq_mgr_mlx5::poll_and_process_element_rx(uint64_t* p_cq_poll_sn, void* pv_fd
...
+		*m_cq_dbell = htonl(m_cq_cons_index);
+		return cqe;
+
+	case MLX5_CQE_INVALID:
+	default:
+		return NULL; /* No CQE */
+	}
+}
+
+inline volatile struct mlx5_cqe64 *cq_mgr_mlx5::get_cqe64(volatile struct mlx5_cqe64 **cqe_err)
```
Comment (rafiw) [inline src/vma/dev/cq_mgr_mlx5.cpp]: you can use
https://github.com/Mellanox/libvma/blob/master/src/vma/dev/cq_mgr_mlx5.inl#L44

Comment (olgk) [inline src/vma/dev/cq_mgr_mlx5.cpp]: Thanks Rafi, the point was to minimize changes, that usage I'd planned for "VMAPOLL in MLX5 derived classe"s or "dynamic_wqe" commits.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1869/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1315/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1319/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1875/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1876/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1320/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1323/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1880/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1325/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1326/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1881/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1882/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1327/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1883/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1328/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1887/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1889/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1343/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1905/ for details (Mellanox internal link).

**Comment by olgk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1910/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1347/ for details (Mellanox internal link).

**Comment by ilansmith** [comment]
Replaced by #418.

**Pull request closed without merging**